### ISSUE TITLE
One-based indexes in Lua error messages 

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#370]
C API and IPROTO are zero-based and consequently all exceptions and error messages are also zero-based. These error messages look inconsistent with one-based Lua API (#347).

```
tarantool> box.space.x:update(1,{{'#',1,0}})
---
- error: 'Field 0 UPDATE error: cannot delete 0 fields'
...
```

**Comment by rtsisyk** [comment]
Mostly fixed by 5fdbc45c07dfbff0ebc1b604e467b510f076daaf

**Comment by kostja** [comment]
This issue and its test is fixed. We have gh-709, gh-616, gh-454 to cover the rest.
